src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -27,3 +31,148 @@ CUDAEmitter::CUDAEmitter(GPUPrimitiveEmitter* emitter)
     : m_primitive_emitter(emitter)
 {
 }
+
+size_t CUDAEmitter::build_pad(const GPURuntimeContext* ctx,
+                              const std::array<std::string, 2>& dtypes,
+                              const ngraph::Shape& input_shape,
+                              const ngraph::Shape& output_shape,
+                              const ngraph::Shape& padding_below,
+                              const ngraph::Shape& padding_above,
+                              const ngraph::Shape& padding_interior)
+{
+    if (padding_interior.size())
+    {
+        throw std::runtime_error("Interior padding is not yet supported in the GPU transformer.");
+    }
+    std::string hash = "pad_i" + join(input_shape) + "_pb" + join(padding_below) + "_pa" +
```

Comment:
join has an optional second parameter which is the separator. You could use calls like `join(input_shape, "_")` and then skip the replace stuff below.

Reply:
good to know. :)